diff --git a/node_modules/react-native-quick-sqlite/android/build.gradle b/node_modules/react-native-quick-sqlite/android/build.gradle
index 71c0db9..3569b23 100644
--- a/node_modules/react-native-quick-sqlite/android/build.gradle
+++ b/node_modules/react-native-quick-sqlite/android/build.gradle
@@ -82,11 +82,12 @@ android {
     packagingOptions {
       doNotStrip resolveBuildType() == 'debug' ? "**/**/*.so" : ''
       excludes = [
-              "META-INF",
-              "META-INF/**",
-              "**/libjsi.so",
-              "**/libreact_nativemodule_core.so",
-              "**/libturbomodulejsijni.so"
+              "META-INF", "META-INF/**", "**/libjsi.so",
+              "**/libreact_nativemodule_core.so", "**/libturbomodulejsijni.so",
+              "**/libc++_shared.so", "**/libfbjni.so", "**/libjsi.so",
+              "**/libreactnativejni.so", "**/libfolly_json.so",
+              "**/libreanimated.so", "**/libjscexecutor.so", "**/libhermes.so",
+              "**/libfolly_runtime.so", "**/libhermes_executor.so"
       ]
     }
     
